### PR TITLE
SPEC-816

### PR DESCRIPTION
**Pull request by jyemin**
Add tests to authentication specification for connection strings with authMechanism but no username

### CODE REVIEW

**Comment by behackett** [comment]
LGTM

File: source/auth/tests/connection-string.js [hunk 1]
```diff
@@ -132,6 +150,12 @@ describe("connection string", function() {
 
 			assert.equal(url.credential.mechanism, "PLAIN");
 		});
+
+		it("should throw an exception when an authMechanism is specified with no username", function() {
+			assert.throws(function() {
+				parse("mongodb://localhost/foo/?authMechanism=PLAIN");
```
Comment (bjori) [inline source/auth/tests/connection-string.js]: I don't think this extra slash after the dbname is valid and would cause other type of exception

Comment (rozza) [inline source/auth/tests/connection-string.js]: Good catch

Comment (jyemin) [inline source/auth/tests/connection-string.js]: Done

**Comment by bjori** [comment]
lgtm

**Pull request merged**